contracts/manager/BaseManager.sol

```diff
@@ -258,9 +258,12 @@ contract BaseManager is MutualUpgrade {
         mutualUpgrade(operator, methodologist)
     {
         require(protectedModules[_module].isProtected, "Module not protected");
-        require(isAdapter[_adapter], "Adapter does not exist");
         require(!protectedModules[_module].authorizedAdapters[_adapter], "Adapter already authorized");
 
+        if (!isAdapter[_adapter]) {
```

Comment:
Bit of a nit: But we can maybe DRY this up a bit more by making an internal function that does the state updates. That can be used here and in `protectModule`.

Reply:
331a399